From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Istvan Zoltan Ballok <istvan.zoltan.ballok@sap.com>
Date: Tue, 7 Jan 2025 13:40:10 +0100
Subject: [PATCH 11/29] Configure a controller runtime logger in the unit test

While debugging the unit test and spending more than 30 seconds in breakpoints I
noticed that the controller runtime's logger emits a warning:

> [controller-runtime] log.SetLogger(...) was never called; logs will not be displayed.
> Detected at:%s%s

https://github.com/kubernetes-sigs/controller-runtime/blob/v0.19.2/pkg/log/log.go#L65

By configuring a logger for the controller runtime in the unit test, this
warning can be avoided and the logs that e.g. the Garbage Collector emits are
shown in the standard out.
---
 .../controller/garbagecollector/reconciler_test.go          | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/pkg/resourcemanager/controller/garbagecollector/reconciler_test.go b/pkg/resourcemanager/controller/garbagecollector/reconciler_test.go
index afb06be54d..60f29d00b5 100644
--- a/pkg/resourcemanager/controller/garbagecollector/reconciler_test.go
+++ b/pkg/resourcemanager/controller/garbagecollector/reconciler_test.go
@@ -8,6 +8,7 @@ import (
 	"context"
 	"time"
 
+	"github.com/go-logr/logr"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 	appsv1 "k8s.io/api/apps/v1"
@@ -18,10 +19,12 @@ import (
 	testclock "k8s.io/utils/clock/testing"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
 	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/logger"
 	"github.com/gardener/gardener/pkg/resourcemanager/apis/config"
 	. "github.com/gardener/gardener/pkg/resourcemanager/controller/garbagecollector"
 	"github.com/gardener/gardener/pkg/resourcemanager/controller/garbagecollector/references"
@@ -30,6 +33,7 @@ import (
 var _ = Describe("Collector", func() {
 	var (
 		ctx = context.TODO()
+		log logr.Logger
 
 		c  client.Client
 		gc *Reconciler
@@ -40,6 +44,8 @@ var _ = Describe("Collector", func() {
 	)
 
 	BeforeEach(func() {
+		log = logger.MustNewZapLogger(logger.DebugLevel, logger.FormatText)
+		logf.SetLogger(log.WithName("garbagecollector"))
 		c = fakeclient.NewClientBuilder().WithScheme(kubernetes.SeedScheme).Build()
 		gc = &Reconciler{
 			TargetClient:          c,
